src/Auth/Storage/StorageInterface.php

```diff
@@ -30,10 +30,10 @@ public function read();
     /**
      * Write user record.
      *
-     * @param array $user User record.
+     * @param array|\ArrayAccess $user User record.
      * @return void
      */
-    public function write(array $user);
+    public function write($user);
```

Comment:
This could cause issues for userland extensions. They will be required to update their code now.